src/main/kotlin/org/rust/lang/core/resolve/ref/RsPathReferenceImpl.kt

```diff
@@ -128,3 +126,40 @@ fun resolvePath(path: RsPath, lookup: ImplLookup = ImplLookup.relativeTo(path)):
 private fun resolveAssocTypeBinding(trait: RsTraitItem, binding: RsAssocTypeBinding): RsTypeAlias? =
     collectResolveVariants(binding.referenceName) { processAssocTypeVariants(trait, it) }
         .singleOrNull() as? RsTypeAlias?
+
+/** Resolves a reference through type aliases */
+fun RsPathReference.throughResolve(): RsElement? =
+    advancedThroughResolve()?.element
+
+/** Resolves a reference through type aliases */
+fun RsPathReference.advancedThroughResolve(): BoundElement<RsElement>? {
```

Comment:
Maybe `deepResolve` is better?

Reply:
I hesitated a long time with the choice, but ok, will be `deepResolve` :)